src/components/Dashboard/Dashboard.tsx

```diff
@@ -66,8 +65,20 @@ export default function Dashboard() {
         const files: File[] = [];
         snap.forEach(child => {
           const data = child.val();
-          if (!showHidden && data.hidden) return;
           const key = child.key;
+          if (!('language' in child.val())) {
```

Comment:
you shouldn't need to edit this file - update `useUpdateUserDashboard` instead (as I did here):

https://github.com/cpinitiative/ide/pull/112/files#diff-a87f1442fe227abc551c4f049e9a5968a64132a0560604d25b0b8cb64268805c

Reply:
The problem with this is that it only updates the language on the dashboard whenever you open the file. This is fine for the most part since all files open on creation. However, this means that previous files which have already been made will not have their languages displayed on the dashboard until the file is opened. I can just have them display their language as "Unknown" until they are opened, but the approach I used should update the language whenever a user opens their dashboard, not each individual file, meaning that all previous files will automatically have their languages set.

Reply:
maybe the if statement should be removed? the language of the file can change, and when it does it should be reflected on the dashboard

Reply:
thats a good point, I got rid of the if statement